costmap_2d/plugins/static_layer.cpp

```diff
@@ -276,6 +282,11 @@ void StaticLayer::updateBounds(double robot_x, double robot_y, double robot_yaw,
       return;
   }
 
+  // if the origin has not changed, skip the re-painting, since we cannot add
+  // any new information
+  if(!has_updated_data_)
+    return;
```

Comment:
I'm not convinced this is right:

- If map_received_ is true, then has_updated_data is set to true, so that case is OK
- but what about has_extra_bounds_ - that is handled in the base class - so it cannot be setting has_updated_data_. I think not doing the updates below when has_extra_bounds_ is true means that things like costmap clearing may not function properly?

I think this extra check and return should be removed.

Reply:
good catch... writing 
```cpp
  if(!has_updated_data_ && !has_extra_bounds_)
    return;
```

would solve this. But I honestly don't understand the logic behind `useExtraBounds` for this layer: Its never set and never used... but thats maybe too much for this PR

Reply:
I believe "useExtraBounds" is only called in clearing the cost maps? It forces data to be copied over? I think removing this extra if/return statement will suffice.

Reply:
Ah yes, you are right, its triggered from clear-costmap-recovery. 

We could remove the if/return but the idea here is to stop re-rendering the entire local costmap if, we aren't actually moving (and thus if we don't have additional info from the static-layer). I've tested the `!has_updated_data_ && !has_extra_bounds_` case and it seems to work. But lastly its your call - shall I drop it?

Reply:
I take it as a yes